### DISCUSSION TITLE
Can not download anything, from any site

### DISCUSSION THREAD

**Discussion by 3ifish** [aclap-dev/video-downloadhelper#1369] [Help and Q&A] [answered]
### Your question

After a short and happy relationship with Video DownloadHelper it has stopped working for me, consistently giving me "... failed at downloading that specific media ..." messages.  I see a number of people with what might appear to be similar failure to download media issues, though I am willing to admit my problem may be me.
 
I've tried using different servers, and different quality settings where the site allows
I've tried with and without AV and Firewall protection.  
I have tested various sites with the same results, and I have tried downloading media sites that I was previously able to download.  NordVPN has been kept on, both when VDH worked and now (for I hope obvious reasons).

Sites tested:
- popcornmovies.to/home
- https://www.braflix.is/
- https://www.vidbinge.com/
- https://fmovies-hd.to/home/
- https://soaper.tv/
- https://bflix.sh/home/

Thanks in advance for looking.

fiiish

### Addon & CoApp version and details

version: 9.1.1.0
target: google
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"8c3c95a6a0c54a54"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "http_media_download_strategy": "inbrowser",
    "used_history_button": true,
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "C:\\Users\\XXXX\\dwhelper"
        }
    },
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": true,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": true,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": true,
    "open_count_store": 1,
    "successfull_dl": 1
}

**Comment by Mattmjs** [comment]
Ok I can see that there are problems depending on the source of the video from these sites. On the ones that fail & don't work
VDH gives the generic we couldn't download this message with a report button. But using ffmpeg directly some of these sources
return a 403 error. Can't do anything on these. I hope there aren't more sites you have trouble with.
I'll give a rundown of each site.

braflix
Golf server, fails as browser download, companion app download works. Right click save video as works.
Oscar server, works but throttled.
Other servers don't work.

vidbinge
Video sources - WHVX, click on this option to select Astra which works.
Video sources - EE3, works.
Autoembed English server, works but throttled.
Video sources - SoaperTv, fails in VDH. Works directly in ffmpeg. Needs a fix.
Other video sources don't work.

fmovies
Premium server, fails in VDH, works directly in ffmpeg. Needs a fix.
Other servers don't work.

Soaper
Already tested, doesn't work in VDH but works directly in ffmpeg. Needs a fix.  

bflix
Vidstream server, works but throttled. Other servers don't work.

**Comment by 3ifish** [comment]
Many (10^6) thanks for your assistance. 

Being very ignorant of the inner working of VDH it seems that when you write re. soaper "Doesn't work in VDH but works directly in ffmpeg", that would mean that I can use this directly in ffmpeg: https://soaper.tv/episode_5XgR0W4kbe.html (if this is correct ffmeg is not working for me)?

Sorry to be so dense.

fiiish

**Reply by Mattmjs** [reply, marked as answer]
No, ffmpeg cannot work that way. It needs a url that has the .m3u8 extension. Or a url that has a .mpd extension.
This is what VDH detects when you go to a page that has a video. You can see the media information by clicking the 3 dots
next to download. Look down to details and click that. There is also a copy url in the drop down menu.

The basic operation is, **ffprobe "video-domain-url.m3u8"** to see the formats.
**ffmpeg -i "video-domain-url.m3u8" -codec: copy "name of your file.mp4"**   to download, only if one video and audio format.

If there are multiple video and audio formats, I've highlighted the extra parameters in bold. The numbers for your formats could be
different to that.

ffmpeg -i "video-domain-url.m3u8" -codec: copy **-map 0:1 -map 0:5** "name of your file.mp4"

**Reply by 3ifish** [reply]
Got it, and thanks again for the help.

fiiish

**Reply by GrampaWildWilly** [reply]
If you are really serious about learning how to use ffmpeg, take a look at this:

https://groups.google.com/g/video-downloadhelper-q-and-a/c/oYDFx367MSY

**Reply by 3ifish** [reply]
Thanks for the info, I'll see how it works out for me. 

Regards

fiiish

**Reply by GrampaWildWilly** [reply]
I have posted a number of examples in this forum of the application of the principles included in that Google Groups tutorial.  I tried to mention the term `Plan B` in those examples, so that would be your search term here.

**Reply by 3ifish** [reply]
@grampawildwilly

Thanks. In regard to nothing, my wife and I just tested positive for Covid (1st time) and I'm doing nothing that requires any memory retention.  I'll let you know how effective ffmpeg has been for me shortly after what I understand is the usual 8–10 day "feeling like cr*p" period is over.

Again, thank you for the additional follow-up, it is very much appreciated.

fiiish

**Reply by GrampaWildWilly** [reply]
Best wishes to you both for a speedy recovery.

**Comment by xezoxeon** [comment]
I get this message

Download failed


Sadly we failed at downloading that specific media. We try to support as many websites as possible, so it would help a lot if you could report that error (it's anonymous!).
